internal/kubectl/kubectl.go

```diff
@@ -37,6 +38,135 @@ const (
 	mimirURL         = "http://mimir.tester:9009/mimir"
 )
 
+type FileEntry struct {
+	RelPath string
+	Key     string
+}
+
+func collectFileEntries(root string, exts map[string]bool) ([]FileEntry, error) {
```

Comment:
Please add a comment explaining the purpose, inputs and outputs for this function.